src/selectors/resolveNotePath.ts

```diff
@@ -0,0 +1,45 @@
+import Path from '../@types/Path'
+import State from '../@types/State'
+import ThoughtId from '../@types/ThoughtId'
+import appendToPath from '../util/appendToPath'
+import head from '../util/head'
+import findDescendant from './findDescendant'
+import { anyChild } from './getChildren'
+import getThoughtById from './getThoughtById'
+
+interface NoteKeyResult {
+  noteKey: string
+  noteId?: ThoughtId
+}
+
+/** Resolves note key and note id by checking for note thoughts. */
+export const resolveNoteKey = (state: State, thoughtId: ThoughtId): NoteKeyResult => {
+  const parentId = getThoughtById(state, thoughtId)?.parentId ?? null
+
+  // the id of the thought's =note or the parent's =children/=note
+  const noteId = findDescendant(state, thoughtId, '=note') || findDescendant(state, parentId, ['=children', '=note'])
+
+  if (!noteId) return { noteKey: '=note' }
+
+  const noteThought = getThoughtById(state, noteId)
+
+  if (noteThought?.pending) return { noteKey: '=note' }
+
+  const notePathId = findDescendant(state, noteId, '=path')
+  const noteKey = anyChild(state, notePathId)?.value ?? '=note'
+  return {
+    noteKey,
+    ...(notePathId ? {} : { noteId }),
```

Comment:
This special case is regrettable, and doesn't totally fit the semantics of the function, but I can't think of an alternative at the moment. We can keep it for the sake of getting this merged 👍